Sources/FoundationEssentials/Data/Data+Writing.swift

```diff
@@ -75,7 +75,7 @@ private func writeToFileDescriptorWithProgress(_ fd: Int32, data: Data, reportPr
         }
     }
 
-    var nextRange = 0..<length
+    var nextRange = data.startIndex..<data.startIndex.advanced(by: length)
```

Comment:
Nit: I can't comment there, but on line 90 I think it's safer to write it as `index.advanced(:)` instead of adding the raw indices? I guess it doesn't make actual difference, so it's just a style nitpick which I'm ok with leaving it asis right now

Reply:
I think you're right, but since Data uses int indexes it's functionally identical. If Data ever switched its index type (which it can't do anyway) then it would just end up being a compile error. So it's a fairly benign nitpick.